docs/features/pouch_with_plugin.md

```diff
@@ -1,148 +1,161 @@
+* post-update container point, at this point you can change the update config into the container's env.
+
+Above five points are organized by container plugins, defined as follow:
+
+```
 // ContainerPlugin defines places where a plugin will be triggered in container lifecycle
 type ContainerPlugin interface {
-  // PreCreate defines plugin point where receives an container create request, in this plugin point user
-  // could change the container create body passed-in by http request body
-  PreCreate(io.ReadCloser) (io.ReadCloser, error)
-
-  // PreStart returns an array of priority and args which will pass to runc, the every priority
-  // used to sort the pre start array that pass to runc, network plugin hook always has priority value 0.
-  PreStart(interface{}) ([]int, [][]string, error)
-
-  //NetworkGenericParams accepts the container id and env of this container and returns the priority of this endpoint
-  // and if this endpoint should enable resolver and a map which will be used as generic params to create endpoints of
-  // this container
-  PreCreateEndpoint(string, []string) (priority int, disableResolver bool, genericParam map[string]interface{})
+	// PreCreate defines plugin point where receives a container create request, in this plugin point user
+	// could change the container create body passed-in by http request body
+	PreCreate(*types.ContainerCreateConfig) error
+
+	// PreStart returns an array of priority and args which will pass to runc, the every priority
+	// used to sort the pre start array that pass to runc, network plugin hook always has priority value 0.
+	PreStart(interface{}) ([]int, [][]string, error)
+
+	// PreCreateEndpoint accepts the container id and env of this container and returns the priority of this endpoint
+	// and if this endpoint should enable resolver and a map which will be used as generic params to create endpoints of
+	// this container
+	PreCreateEndpoint(string, []string) (priority int, disableResolver bool, genericParam map[string]interface{})
+
+	// PreUpdate defines plugin point where receives a container update request, in this plugin point user
+	// could change the container update body passed-in by http request body
+	PreUpdate(io.ReadCloser) (io.ReadCloser, error)
+
+	// PostUpdate called after update method successful,
+	// the method accepts the rootfs path and envs of container
+	PostUpdate(string, []string) error
 }
-
 ```
 
-These two Plugin symbols will be fetch by name `DaemonPlugin` and `ContainerPlugin` from shared object file like this:
+### daemon plugin
+
+* pre-start daemon point, at this point you can start assist processes like network plugins and dfget proxy which need by pouchd and whose life cycle is the same as pouchd.
+* pre-stop daemon point, at this point you can stop the assist processes gracefully, but the trigger of this point is not a promise, because pouchd may be killed by SIGKILL.
```

Comment:
```suggestion
* pre-stop daemon point, at this point you can stop the assistant processes gracefully, but the trigger of this point is not a promise, because pouchd may be killed by SIGKILL.
```